scylla-cdc/src/consumer.rs

```diff
@@ -0,0 +1,485 @@
+        let mut ttl = None;
+
+        let data_count =
+            schema.mapping.len() + schema.deleted_mapping.len() + schema.deleted_el_mapping.len();
+        let mut data: Vec<Option<CqlValue>> = Vec::with_capacity(data_count);
+
+        for (i, column) in row.columns.into_iter().enumerate() {
+            if i == schema.stream_id {
+                stream_id = column.unwrap().into_blob().unwrap();
+            } else if i == schema.time {
+                time = column.unwrap().as_uuid().unwrap();
+            } else if i == schema.batch_seq_no {
+                batch_seq_no = column.unwrap().as_int().unwrap();
+            } else if i == schema.end_of_batch {
+                end_of_batch = column.unwrap().as_boolean().unwrap()
+            } else if i == schema.operation {
+                operation = OperationType::try_from(column.unwrap().as_tinyint().unwrap()).unwrap();
+            } else if i == schema.ttl {
+                ttl = column.map(|ttl| ttl.as_bigint().unwrap());
+            } else {
+                data.push(column);
+            }
+        }
+
+        CDCRow {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            data,
+            schema,
+        }
+    }
+
+    /// Allows to get a value from the column that corresponds to the logged table.
+    /// Returns None if the value is null.
+    /// Panics if the column does not exist in this table.
+    /// To check if such column exists, use column_exists() method.
+    pub fn get_value(&self, name: &str) -> &Option<CqlValue> {
+        self.schema
+            .mapping
+            .get(name)
+            .map(|id| &self.data[*id])
+            .unwrap()
+    }
+
+    /// Allows to get info if a value was deleted in this operation.
+    /// Panics if the column does not exist in this table
+    /// or the column is a part of primary key (because these values can't be deleted).
+    /// To check if such column exists, use column_deletable() method.
+    pub fn is_value_deleted(&self, name: &str) -> bool {
+        self.schema
+            .deleted_mapping
+            .get(name)
+            .map(|id| self.data[*id].is_some())
+            .unwrap()
+    }
+
+    /// Allows to get deleted elements from a collection.
+    /// Returns empty slice if the value is null.
+    /// Panics if the column does not exist in this table or is not a collection.
+    /// To check if such column exists, use collection_exists() method.
+    pub fn get_deleted_elements(&self, name: &str) -> &[CqlValue] {
+        let val = self
+            .schema
+            .deleted_el_mapping
+            .get(name)
+            .map(|id| self.data[*id].as_ref().map(|val| val.as_set().unwrap()))
+            .unwrap();
+        match val {
+            Some(vec) => vec,
+            None => &[],
+        }
+    }
+
+    pub fn column_exists(&self, name: &str) -> bool {
```

Comment:
I think the newly added methods belong to `CDCRowSchema` rather than `CDCRow`, so please move them there.

Reply:
I thought that these functions are for the consumer so that he can use them, for example, to debug. As far as I know, the consumer isn't going to see any `CDCRowSchema`, because this is an internal struct that is used to create new `CDCRow` objects. If these functions were moved to `CDCRowSchema`, they would be useless in my opinion.

Reply:
Personally, I see nothing wrong with exposing the `CDCRowSchema` to the user, (for example through `CDCRow::get_schema()` or something) and I think that those methods make more sense to be there, but maybe it will make the interface harder to use... I'm not sure, so I won't insist.